packages/api/tests/models/granules/test-granule-generateGranuleRecord.js

```diff
@@ -141,21 +120,23 @@ test('generateGranuleRecord() builds a failed granule record', async (t) => {
     Error: 'error',
     Cause: new Error('error'),
   };
+  const workflowStatus = 'failed';
   const record = await granuleModel.generateGranuleRecord({
-    s3: t.context.fakeS3,
     granule,
     message: granuleFailure,
     executionUrl,
-    provider,
+    provider: provider.id,
+    files: granule.files,
     collectionId,
     workflowStartTime,
-    workflowStatus: 'failed',
+    workflowStatus,
+    status: getGranuleStatus(workflowStatus, granule),
     error,
   });
 
   t.deepEqual(
     record.files,
-    granule.files.map((file) => granuleFileToRecord(file, provider))
+    granule.files
```

Comment:
Before, they were being built by buildDatabaseFiles inside the call, and now they are just being passed along. (For granules from messages the files pass to this still goes through the same function so no change in what's being stored).  I suppose for the API they have some freedom to do what they want.  I can take out this check since it's not doing much if you think that's misleading?

Reply:
No, I think it's fine, I mostly just wanted to check my understanding. I like it much better now, in fact